src/test/java/uk/gov/pay/directdebit/mandate/services/MandateServiceTest.java

```diff
@@ -38,99 +48,188 @@
+                .toEntity();
+        when(mockedMandateDao.findByExternalId(mandate.getExternalId())).thenReturn(Optional.of(mandate));
+        Mandate newMandate = service.confirmedDirectDebitDetailsFor(mandate.getExternalId());
+        assertThat(newMandate.getId(), is(notNullValue()));
+        assertThat(newMandate.getExternalId(), is(mandate.getExternalId()));
+        assertThat(newMandate.getReturnUrl(), is(mandate.getReturnUrl()));
+        assertThat(newMandate.getGatewayAccount(), is(mandate.getGatewayAccount()));
+        assertThat(newMandate.getReference(), is(mandate.getReference()));
+        assertThat(newMandate.getPayer(), is(mandate.getPayer()));
+        assertThat(newMandate.getType(), is(mandate.getType()));
+        assertThat(newMandate.getState(), is(SUBMITTED));
+        assertThat(newMandate.getCreatedDate(), is(mandate.getCreatedDate()));
+    }
 
-        Transaction transaction = TransactionFixture
-                .aTransactionFixture()
-                .withState(PENDING_DIRECT_DEBIT_PAYMENT)
+    @Test
+    public void mandatePendingFor_shouldRegisterAMandatePendingEvent() {
+        Mandate mandate = MandateFixture
+                .aMandateFixture()
+                .withState(SUBMITTED)
                 .toEntity();
 
-        PaymentRequestEvent event = PaymentRequestEventFixture.aPaymentRequestEventFixture().toEntity();
+        service.mandatePendingFor(mandate);
 
-        when(mockedPaymentRequestEventService.findBy(transaction.getPaymentRequest().getId(), PaymentRequestEvent.Type.MANDATE, MANDATE_PENDING))
-                .thenReturn(Optional.of(event));
+        verify(mockedPaymentRequestEventService).registerMandatePendingEventFor(mandate);
+        assertThat(mandate.getState(), is(PENDING));
+    }
 
-        PaymentRequestEvent foundEvent = service.findMandatePendingEventFor(transaction).get();
+    @Test
+    public void shouldThrowWhenSubmittingTheSameDetailsSeveralTimes_andCreateOnlyOneEvent_whenConfirmingDirectDebitDetails() {
+        Mandate mandate = MandateFixture
+                .aMandateFixture()
+                .withState(AWAITING_DIRECT_DEBIT_DETAILS)
+                .toEntity();
 
-        assertThat(foundEvent, is(event));
+        when(mockedMandateDao.findByExternalId(mandate.getExternalId())).thenReturn(Optional.of(mandate));
+
+        // first call with confirmation details
+        service.confirmedDirectDebitDetailsFor(mandate.getExternalId());
+        thrown.expect(InvalidStateTransitionException.class);
+        thrown.expectMessage("Transition DIRECT_DEBIT_DETAILS_CONFIRMED from state SUBMITTED is not valid");
+
+        // second call with same details that should throw and prevent adding a new event
+        Mandate newMandate = service.confirmedDirectDebitDetailsFor(mandate.getExternalId());
+        assertThat(newMandate.getState(), is(SUBMITTED));
+        verify(mockedPaymentRequestEventService, times(1)).registerDirectDebitConfirmedEventFor(newMandate);
```

Comment:
That’s not quite what I meant about adding blank lines. I was requesting something like this:

```java
when(mockedMandateDao.findByExternalId(mandate.getExternalId())).thenReturn(Optional.of(mandate));
// first call with confirmation details
service.confirmedDirectDebitDetailsFor(mandate.getExternalId());

thrown.expect(InvalidStateTransitionException.class);
thrown.expectMessage("Transition DIRECT_DEBIT_DETAILS_CONFIRMED from state SUBMITTED is not valid");
// second call with same details that should throw and prevent adding a new event
Mandate newMandate = service.confirmedDirectDebitDetailsFor(mandate.getExternalId());

assertThat(newMandate.getState(), is(SUBMITTED));
verify(mockedPaymentRequestEventService, times(1)).registerDirectDebitConfirmedEventFor(newMandate);
```

As that makes it clearer that it’s the second call that throws the exception.